### DISCUSSION TITLE
Scrobber plugin does not work on mac osx

### DISCUSSION THREAD

**Discussion by guysoft** [audacious-media-player/audacious#111] [Q&A] [answered]
Hey,
The last.fm scrobber plugin does not work on mac osx.
The issue is that the button to activation does not bring up the popup that I see on linux.
I tried looking in stdout/stderror but there is no message there.
What can I use to get a debug output?

I am running Audacous on mac using brew install audacious. Mac M2 OSX 13

Is there a way to debug what is going on?
Thanks!

**Comment by radioactiveman** [comment]
@guysoft: The output of "audacious -VV" may be helpful. However I was able to reproduce this myself on macOS.

@jlindgren90: This is an interesting issue. :) The callback function `permission_checker_thread()` is never called on macOS. It should be triggered by `g_timeout_add()` every 250 ms. My first guess was that no GLib mainloop is running on macOS. But this works with a Qt-only build of Audacious on Linux.

A solution would be to use our `QueuedFunc` implementation from `libaudcore`. I have tested it successfully on macOS. But do you have an assumption why the GLib approach works on Linux? Which mainloop handles the GLib events?

https://github.com/audacious-media-player/audacious-plugins/blob/b6672385e89d13cf8b1c56096cb6c29195b0b38a/src/scrobbler2/config_window.cc#L77

---

```c++
#include <glib.h>
#include <QApplication>

// Why/how is this called on Linux without a GLib mainloop?
static int timer_cb(void *)
{
    g_print("%s\n", __FUNCTION__);
    return true;
}

int main(int argc, char **argv)
{
    QApplication app(argc, argv);

    g_timeout_add(250, timer_cb, nullptr);

    return app.exec();
}
```

**Reply by jlindgren90** [reply]
Qt does use a GLib main loop on Linux for compatibility, I think. And probably doesn't on Mac. We ran into similar issues on Windows a while back.

Using our QueuedFunc is the best solution, I think.

**Reply by radioactiveman** [reply, marked as answer]
Should be fixed now with https://github.com/audacious-media-player/audacious-plugins/commit/895e2c3019cc50ec3dd54cd9385c42a35d604f8d for Audacious 4.4.

**Comment by guysoft** [comment]
Can confirm it works now Audacious 4.4
